src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -0,0 +1,203 @@
+  public VersionBound commonProtosVersion(String language) {
+    return getWithDefault(commonProtosVersion(), language);
+  }
+
+  /** The version the client library package. E.g., "0.14.0". Configured per language. */
+  public VersionBound packageVersion(String language) {
+    return getWithDefault(packageVersion(), language);
+  }
+
+  /**
+   * The base name of the client library package. E.g., "google-cloud-logging-v1". Configured per
+   * language.
+   */
+  public String packageName(String language) {
+    return getWithDefault(packageName(), language);
+  }
+
+  /** A single-word short name of the API. E.g., "logging". */
+  public abstract String shortName();
+
+  /** The major version of the API, as used in the package name. E.g., "v1". */
+  public abstract String apiVersion();
+
+  /** The path to the API protos in the googleapis repo. */
+  public abstract String protoPath();
+
+  /** The author of the client library. */
+  public abstract String author();
+
+  /** The email of the author of the client library. */
+  public abstract String email();
+
+  /** The homepage of the client library. */
+  public abstract String homepage();
+
+  /** The name of the license of the client library. */
+  public abstract String license();
+
+  private static Builder newBuilder() {
+    return new AutoValue_PackageMetadataConfig.Builder();
+  }
+
+  @AutoValue.Builder
+  protected abstract static class Builder {
+    abstract Builder gaxVersion(Map<String, VersionBound> val);
+
+    abstract Builder protoVersion(Map<String, VersionBound> val);
+
+    abstract Builder commonProtosVersion(Map<String, VersionBound> val);
+
+    abstract Builder packageName(Map<String, String> val);
+
+    abstract Builder packageVersion(Map<String, VersionBound> val);
+
+    abstract Builder shortName(String val);
+
+    abstract Builder apiVersion(String val);
+
+    abstract Builder protoPath(String val);
+
+    abstract Builder author(String val);
+
+    abstract Builder email(String val);
+
+    abstract Builder homepage(String val);
+
+    abstract Builder license(String val);
+
+    abstract PackageMetadataConfig build();
+  }
+
+  /** Creates an PackageMetadataConfig with no content. Exposed for testing. */
+  @VisibleForTesting
+  public static PackageMetadataConfig createDummyPackageMetadataConfig() {
+    return newBuilder()
+        .gaxVersion(ImmutableMap.<String, VersionBound>of())
+        .protoVersion(ImmutableMap.<String, VersionBound>of())
+        .packageName(ImmutableMap.<String, String>of())
+        .commonProtosVersion(ImmutableMap.<String, VersionBound>of())
+        .packageVersion(ImmutableMap.<String, VersionBound>of())
+        .shortName("")
+        .apiVersion("")
+        .protoPath("")
+        .author("")
+        .email("")
+        .homepage("")
+        .license("")
+        .build();
+  }
+
+  @SuppressWarnings("unchecked")
+  public static PackageMetadataConfig createFromFile(Path packageConfigPath) throws IOException {
```

Comment:
I think the actual place where file I/O should be handled is at the top level; transformers accept an in-memory data model as input.